pkg/server/server.go

```diff
@@ -46,46 +51,109 @@ var tracer = otel.Tracer("openfga/pkg/server")
 type Server struct {
 	openfgapb.UnimplementedOpenFGAServiceServer
 
-	logger              logger.Logger
-	datastore           storage.OpenFGADatastore
-	encoder             encoder.Encoder
-	transport           gateway.Transport
-	config              *Config
+	logger                 logger.Logger
+	datastore              storage.OpenFGADatastore
+	encoder                encoder.Encoder
+	transport              gateway.Transport
+	resolveNodeLimit       uint32
+	changelogHorizonOffset int
+	listObjectsDeadline    time.Duration
+	listObjectsMaxResults  uint32
+	experimentals          []ExperimentalFeatureFlag
+
 	optimizeListObjects bool
 }
 
-type Dependencies struct {
-	Datastore    storage.OpenFGADatastore
-	Logger       logger.Logger
-	Transport    gateway.Transport
-	TokenEncoder encoder.Encoder
+type OpenFGASeviceV1Option func(s *Server)
+
+func WithDatastore(ds storage.OpenFGADatastore) OpenFGASeviceV1Option {
+	return func(s *Server) {
+		s.datastore = ds
+	}
+}
+
+func WithLogger(l logger.Logger) OpenFGASeviceV1Option {
+	return func(s *Server) {
+		s.logger = l
+	}
+}
+
+func WithTokenEncoder(encoder encoder.Encoder) OpenFGASeviceV1Option {
```

Comment:
unused, please remove :D

Reply:
It's an option needed to configure the encoder, and services embedding an OpenFGA server (e.g. Auth0 FGA) use it to provide alternative token encoder mechanisms. In Auth0 FGA we use an encrypted token encoder instead of the base64 one.